api/tests/unit/application/assessments/assessment-controller-get-solution_test.js

```diff
@@ -0,0 +1,158 @@
+      sandbox.stub(solutionRepository, 'get');
+      sandbox.stub(solutionSerializer, 'serialize');
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it('should return 404 when the assessment is not found', () => {
+      // given
+      assessmentRepository.get.resolves();
+
+      // when
+      const promise = assessmentController.getAssessmentSolution({ params: { id: 13465 } }, replyStub);
+
+      // then
+      return promise.then(() => {
+        sinon.assert.calledWith(assessmentRepository.get, 13465);
+        sinon.assert.calledOnce(Boom.notFound);
+        sinon.assert.calledWith(replyStub, boomResponseForNotFound);
+      });
+    });
+
+    it('should return 202 when the assessment is found but not completed yet', () => {
+      // given
+      assessmentRepository.get.resolves(new Assessment({ id: 13465, estimatedLevel: null, pixScore: null }));
+
+      // when
+      const promise = assessmentController.getAssessmentSolution({ params: { id: 13465 } }, replyStub);
+
+      // then
+      return promise.then(() => {
+        sinon.assert.calledWith(assessmentRepository.get, 13465);
+        sinon.assert.calledWith(Boom.conflict, 'Cette évaluation n\'est pas terminée.');
+        sinon.assert.calledWith(replyStub, boomResponseForConflict);
+      });
+    });
+
+    context('when a repository is on error', () => {
+      it('should log the error for the assessmentRepository', () => {
+        // given
+        const error = new Error('Database locked');
+        assessmentRepository.get.rejects(error);
+
+        // when
+        const promise = assessmentController.getAssessmentSolution({ params: { id: 13465 } }, replyStub);
+
+        // then
+        return promise.then(() => {
+          sinon.assert.calledWith(logger.error, error);
+          sinon.assert.calledOnce(Boom.badImplementation);
+          sinon.assert.calledWith(replyStub, boomResponseForbBadImplementation);
+        });
+      });
+
+      it('should log the error for the answerRepository', () => {
+        // given
+        const error = new Error('Database locked');
+        answerRepository.get.rejects(error);
+        assessmentRepository.get.resolves(new Assessment({ id: 13465, estimatedLevel: 1, pixScore: 12 }));
+
+        // when
+        const promise = assessmentController.getAssessmentSolution({ params: { id: 13465 } }, replyStub);
+
+        // then
+        return promise.then(() => {
+          sinon.assert.calledWith(logger.error, error);
+          sinon.assert.calledOnce(Boom.badImplementation);
+          sinon.assert.calledWith(replyStub, boomResponseForbBadImplementation);
+        });
+      });
+    });
+
+    context('when the assessment exists and is completed', () => {
+      beforeEach(() => {
+        assessmentRepository.get.resolves(new Assessment({ id: 13465, estimatedLevel: 1, pixScore: 12 }));
+      });
+
+      it('should return an 404 error when the answer is not found', () => {
```

Comment:
Tu peux créer un `context('when the answer is not found', () => {` pour rendre cela plus visible.
Ça m'a perdu dans la lecture du test ligne 136 qui mérite aussi un `context('when the answer is found', () => {`